docs/pypi-dependencies.md

```diff
@@ -44,6 +45,78 @@ hermetic host python interpreter on this platform. Linux and OSX users should se
 difference.
 ```
 
+(bazel-downloader)=
+#### Using `pypi.install` extension
+
+:::{warning}
+The APIs here have been well tested but may still change in backwards
+incompatible ways if major design flaws are found.
+:::
+
+`pypi.install` provides almost a drop-in replacement for the `pip.parse` extension:
+```starlark
+pypi = use_extension("@rules_python//python/extensions:pypi.bzl", "pypi")
+pypi.install(
+    hub_name = "my_deps",
+    python_version = "3.11",
+    requirements_lock = "//:requirements_lock_3_11.txt",
+)
+use_repo(pypi, "my_deps")
+```
+
+The `bzlmod` `pypi.install` call supports pulling information from `PyPI` (or a
+compatible mirror) and it will ensure that the [bazel
+downloader][bazel_downloader] is used for downloading the wheels or the sdists
+in the requirements file. This allows the users to use the [credential
+helper](#credential-helper) to authenticate with the mirror and it also ensures
+that the distribution downloads are cached.
+It also avoids using `pip` altogether and results in much faster dependency
+fetching.
+
+See the {gh-path}`examples/bzlmod/MODULE.bazel` example for more info.
+
+:::{warning}
+For now there is a limitation that the lock
+files consumed by the `pypi.install` extension must have hashes, similarly to
+all our examples. Hence, the `requirements.txt` files produced by `pip lock` or
+by `uv pip compile` but without generating hashes will fail.
+:::
+
+Note, when using this feature during the `pip` extension evaluation you will see the accessed indexes similar to below:
+```console
+Loading: 0 packages loaded
+    currently loading: docs/
+    Fetching module extension pip in @@//python/extensions:pypi.bzl; starting
+    Fetching https://pypi.org/simple/twine/
+```
+
+This does not mean that `rules_python` is fetching the wheels eagerly, but it
+rather means that it is calling the PyPI server to get the Simple API response
+to get the list of all available source and wheel distributions. Once it has
+got all of the available distributions, it will select the right ones depending
+on the `sha256` values in your `requirements_lock.txt` file. The compatible
+distribution URLs will be then written to the `MODULE.bazel.lock` file. Currently
+users wishing to use the lock file with `rules_python` with this feature have
+to set an environment variable `RULES_PYTHON_OS_ARCH_LOCK_FILE=0` which will
+become default in the next release.
+
+Fetching the distribution information from the PyPI allows `rules_python` to
+know which `whl` should be used on which target platform and it will determine
+that by parsing the `whl` filename based on [PEP600], [PEP656] standards. This
+allows the user to configure the behaviour by using the following publicly
```

Comment:
Something is off about what this sentence is introducing and trying to say. I think it's trying to say that these flags can be used to control which wheels are used by a binary?

Reply:
stet on this comment; i see now this was just moving content.